Bump secp256k1-sys to v0.13.0

nymius opened the pull request:
These changes bump the version of the secp256k1-sys internal crate to version 0.13.0.
It includes all the changes from libsecp256k1 at v0.7.0 and a possible fix for the linking issues related to the change the pointers `secp256k1_context_static` and `secp256k1_context_no_precomp` to `const` pointers.

Fixes #878

tcharding commented:
Needs a changelog entry please @nymius. Do we want to cut a `secp256k1 0.33.0` release then @apoelstra?

nymius commented:
I didn't know the purpose of the `./depend/secp256k1.h.patch` file, and I was about to ask but could come up with an answer after some research.

Correct me if I'm wrong, but these patches are intended to avoid definition collision with functions implemented purely in rust.
`secp256k1.h.patch` is formatted using `ed` format and when the `secp256k1-sys` source files get modified, the patch must be updated to keep doing what it is intended to do.
To update the `patch` I ran
```bash
./vendor-libsecp.sh -f v0.7.0
```
, copied the `secp256k1.h` to `secp256k1.h.modified`, interpreted the `ed` format and applied the changes to `secp256k1.h.modified` manually. Created a new diff patch with:
```bash
diff secp256k1.h secp256k1.h.modified > secp256k1.h.patch
```

checked the `patch` was ok by running

```bash
patch --dry-run secp256k1.h < secp256k1.h.patch
```

re-setted the branch to a clean state on top of master, keeping the new patch file and re executed
```bash
./vendor-libsecp.sh -f v0.7.0
```

This time I didn't have to answer to the `Reversed (or previously applied) patch detected! Assume -R? [n]`, because the patch applied successfully.

Now, after doing all this I ran again:
```bash
cargo test --all-features
```

Not hitting any issue, to my surprise.

It seems the offending context functions that changed the ABI in libsecp256k1 v0.7.0 were not even used by `rust-secp256k1`, as the aforementioned patch removed them from the headers file, because rust-secp256k1 implements its own context.

So, there are two options:
- The visibility issue is there, but is not surfaced thanks to rust-secp256k1 removing the lines introducing it.
- There is no visibility issue.
  
I hope the answer isn't that I managed myself to uncover a non-issue and then fixed it. 

When the patch didn't work as intended, it left the library with two duplicated un-reductible symbols that conflicted between them because the native rust one was marked as extern but the libsecp one was not (`SECP256K1_API = Some("")`). After removing this symbol from `build.rs` or making it `SECP256K1_API = Some("extern")`, my guess is LTO was able to reduce this code to the same one, not producing any more errors.

After all, I would ask Matt if the `-fvisibility=<value>` introduced in libsecp256k1 v0.7.0 serves the same purpose of the original `SECP256K1_API = Some("")` change for its use case.

Also, I think all this discussion is relevant to #680, so linking it here.

Finally, I have pushed the changes with the right patches, and removed the SECP256K1_API removal to support what I've explained above.
